File: gencode/java/udmi/schema/Metadata.java

package udmi.schema;

import java.util.Date;
import java.util.HashMap;
import java.util.Map;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;


/**
 * Metadata
 * <p>
 * [Metadata](../docs/specs/metadata.md) is a description about the device: a specification about how the device should be configured and expectations about what the device should be doing. Defined by `metadata.json`
 * 
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
    "timestamp",
    "version",
    "upgraded_from",
    "hash",
    "operation",
    "cloud",
    "system",
    "gateway",
    "discovery",
    "localnet",
    "testing",
    "features",
    "pointset",
    "structure"
})
public class Metadata {

    /**
     * RFC 3339 timestamp UTC the data was generated
     * (Required)
     * 
     */
    @JsonProperty("timestamp")
    @JsonPropertyDescription("RFC 3339 timestamp UTC the data was generated")
    public Date timestamp;
    /**
     * Version of the UDMI schema for this file
     * (Required)
     * 
     */
    @JsonProperty("version")
    @JsonPropertyDescription("Version of the UDMI schema for this file")
    public java.lang.String version;
    /**
     * Original version of the UDMI schema for this file
     * 
     */
    @JsonProperty("upgraded_from")
    @JsonPropertyDescription("Original version of the UDMI schema for this file")
    public java.lang.String upgraded_from;
    /**
     * DEPRECATED
     * 
     */
    @JsonProperty("hash")
    @JsonPropertyDescription("DEPRECATED")
    public java.lang.String hash;
    /**
     * Model Operation
     * <p>
     * 
     * 
     */
    @JsonProperty("operation")
    public CloudModel.ModelOperation operation;
    /**
     * Cloud Model
     * <p>
     * Information specific to how the device communicates with the cloud.
     * 
     */
    @JsonProperty("cloud")
    @JsonPropertyDescription("Information specific to how the device communicates with the cloud.")
    public CloudModel cloud;
    /**
     * System Model
     * <p>
     * High-level system information about the device. [System Model Documentation](../docs/messages/system.md)
     * (Required)
     * 
     */
    @JsonProperty("system")
    @JsonPropertyDescription("High-level system information about the device. [System Model Documentation](../docs/messages/system.md)")
    public SystemModel system;
    /**
     * Gateway Model
     * <p>
     * [Gateway Documentation](../docs/specs/gateway.md)
     * 
     */
    @JsonProperty("gateway")
    @JsonPropertyDescription("[Gateway Documentation](../docs/specs/gateway.md)")
    public GatewayModel gateway;
    /**
     * Discovery Model
     * <p>
     * Discovery target parameters
     * 
     */
    @JsonProperty("discovery")
    @JsonPropertyDescription("Discovery target parameters")
    public DiscoveryModel discovery;
    /**
     * Localnet Model
     * <p>
     * Used to describe device local network parameters
     * 
     */
    @JsonProperty("localnet")
    @JsonPropertyDescription("Used to describe device local network parameters")
    public LocalnetModel localnet;
    /**
     * Testing Model
     * <p>
     * Testing target parameters
     * 
     */
    @JsonProperty("testing")
    @JsonPropertyDescription("Testing target parameters")
    public TestingModel testing;
    /**
     * Testing Model
     * <p>
     * Model of supported features
     * 
     */
    @JsonProperty("features")
    @JsonPropertyDescription("Model of supported features")
    public Map<String, FeatureDiscovery> features;
    /**
     * Pointset Model
     * <p>
     * Pointset representing the abstract system expectation for what the device should be doing, and how it should be configured and operated. This block specifies the expected points that a device holds
     * 
     */
    @JsonProperty("pointset")
    @JsonPropertyDescription("Pointset representing the abstract system expectation for what the device should be doing, and how it should be configured and operated. This block specifies the expected points that a device holds")
    public PointsetModel pointset;
    @JsonProperty("structure")
    public HashMap<String, DiscoveryEvents> structure;

    @Override
    public int hashCode() {
        int result = 1;
        result = ((result* 31)+((this.testing == null)? 0 :this.testing.hashCode()));
        result = ((result* 31)+((this.version == null)? 0 :this.version.hashCode()));
        result = ((result* 31)+((this.structure == null)? 0 :this.structure.hashCode()));
        result = ((result* 31)+((this.cloud == null)? 0 :this.cloud.hashCode()));
        result = ((result* 31)+((this.features == null)? 0 :this.features.hashCode()));
        result = ((result* 31)+((this.system == null)? 0 :this.system.hashCode()));
        result = ((result* 31)+((this.discovery == null)? 0 :this.discovery.hashCode()));
        result = ((result* 31)+((this.upgraded_from == null)? 0 :this.upgraded_from.hashCode()));
        result = ((result* 31)+((this.pointset == null)? 0 :this.pointset.hashCode()));
        result = ((result* 31)+((this.operation == null)? 0 :this.operation.hashCode()));
        result = ((result* 31)+((this.hash == null)? 0 :this.hash.hashCode()));
        result = ((result* 31)+((this.gateway == null)? 0 :this.gateway.hashCode()));
        result = ((result* 31)+((this.localnet == null)? 0 :this.localnet.hashCode()));
        result = ((result* 31)+((this.timestamp == null)? 0 :this.timestamp.hashCode()));
        return result;
    }

    @Override
    public boolean equals(Object other) {
        if (other == this) {
            return true;
        }
        if ((other instanceof Metadata) == false) {
            return false;
        }
        Metadata rhs = ((Metadata) other);
        return (((((((((((((((this.testing == rhs.testing)||((this.testing!= null)&&this.testing.equals(rhs.testing)))&&((this.version == rhs.version)||((this.version!= null)&&this.version.equals(rhs.version))))&&((this.structure == rhs.structure)||((this.structure!= null)&&this.structure.equals(rhs.structure))))&&((this.cloud == rhs.cloud)||((this.cloud!= null)&&this.cloud.equals(rhs.cloud))))&&((this.features == rhs.features)||((this.features!= null)&&this.features.equals(rhs.features))))&&((this.system == rhs.system)||((this.system!= null)&&this.system.equals(rhs.system))))&&((this.discovery == rhs.discovery)||((this.discovery!= null)&&this.discovery.equals(rhs.discovery))))&&((this.upgraded_from == rhs.upgraded_from)||((this.upgraded_from!= null)&&this.upgraded_from.equals(rhs.upgraded_from))))&&((this.pointset == rhs.pointset)||((this.pointset!= null)&&this.pointset.equals(rhs.pointset))))&&((this.operation == rhs.operation)||((this.operation!= null)&&this.operation.equals(rhs.operation))))&&((this.hash == rhs.hash)||((this.hash!= null)&&this.hash.equals(rhs.hash))))&&((this.gateway == rhs.gateway)||((this.gateway!= null)&&this.gateway.equals(rhs.gateway))))&&((this.localnet == rhs.localnet)||((this.localnet!= null)&&this.localnet.equals(rhs.localnet))))&&((this.timestamp == rhs.timestamp)||((this.timestamp!= null)&&this.timestamp.equals(rhs.timestamp))));
    }

}

File: gencode/python/udmi/schema/metadata.py
"""Generated class for metadata.json"""
from .model_cloud import CloudModel
from .model_system import SystemModel
from .model_relationships import RelationshipsModel
from .model_externals import ExternalsModel
from .model_gateway import GatewayModel
from .model_discovery import DiscoveryModel
from .model_localnet import LocalnetModel
from .model_testing import TestingModel
from .model_features import TestingModel
from .model_pointset import PointsetModel
from .events_discovery import DiscoveryEvents


class Metadata:
  """Generated schema class"""

  def __init__(self):
    self.timestamp = None
    self.version = None
    self.upgraded_from = None
    self.hash = None
    self.operation = None
    self.cloud = None
    self.system = None
    self.relationships = None
    self.externals = None
    self.gateway = None
    self.discovery = None
    self.localnet = None
    self.testing = None
    self.features = None
    self.pointset = None
    self.structure = None

  @staticmethod
  def from_dict(source):
    if not source:
      return None
    result = Metadata()
    result.timestamp = source.get('timestamp')
    result.version = source.get('version')
    result.upgraded_from = source.get('upgraded_from')
    result.hash = source.get('hash')
    result.operation = source.get('operation')
    result.cloud = CloudModel.from_dict(source.get('cloud'))
    result.system = SystemModel.from_dict(source.get('system'))
    result.relationships = LinkRelationshipsModel.map_from(source.get('relationships'))
    result.externals = LinkExternalsModel.map_from(source.get('externals'))
    result.gateway = GatewayModel.from_dict(source.get('gateway'))
    result.discovery = DiscoveryModel.from_dict(source.get('discovery'))
    result.localnet = LocalnetModel.from_dict(source.get('localnet'))
    result.testing = TestingModel.from_dict(source.get('testing'))
    result.features = FeatureDiscovery.map_from(source.get('features'))
    result.pointset = PointsetModel.from_dict(source.get('pointset'))
    result.structure = DiscoveryEvents.map_from(source.get('structure'))
    return result

  @staticmethod
  def map_from(source):
    if not source:
      return None
    result = {}
    for key in source:
      result[key] = Metadata.from_dict(source[key])
    return result

  @staticmethod
  def expand_dict(input):
    result = {}
    for property in input:
      result[property] = input[property].to_dict() if input[property] else {}
    return result

  def to_dict(self):
    result = {}
    if self.timestamp:
      result['timestamp'] = self.timestamp # 5
    if self.version:
      result['version'] = self.version # 5
    if self.upgraded_from:
      result['upgraded_from'] = self.upgraded_from # 5
    if self.hash:
      result['hash'] = self.hash # 5
    if self.operation:
      result['operation'] = self.operation # 5
    if self.cloud:
      result['cloud'] = self.cloud.to_dict() # 4
    if self.system:
      result['system'] = self.system.to_dict() # 4
    if self.relationships:
      result['relationships'] = LinkRelationshipsModel.expand_dict(self.relationships) # 2
    if self.externals:
      result['externals'] = LinkExternalsModel.expand_dict(self.externals) # 2
    if self.gateway:
      result['gateway'] = self.gateway.to_dict() # 4
    if self.discovery:
      result['discovery'] = self.discovery.to_dict() # 4
    if self.localnet:
      result['localnet'] = self.localnet.to_dict() # 4
    if self.testing:
      result['testing'] = self.testing.to_dict() # 4
    if self.features:
      result['features'] = FeatureDiscovery.expand_dict(self.features) # 2
    if self.pointset:
      result['pointset'] = self.pointset.to_dict() # 4
    if self.structure:
      result['structure'] = DiscoveryEvents.expand_dict(self.structure) # 2
    return result
